src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -418,6 +422,55 @@ class WorkspaceService(dataSource: DataSource, workspaceDAO: WorkspaceDAO, entit
       // to ensure it is evaluated within the transaction
       RequestComplete(methodConfigurationDAO.list(workspaceNamespace, workspaceName, txn).toList)
     }
+
+  // TODO: replace this stub with the actual expression parser
+  private def resolveInputs( methodConfig: MethodConfiguration, entity: Entity ): String =
+    """{"hello": "world"}"""
+
+  def submitJob(jobDesc: JobDescription, authCookie: HttpCookie): PerRequestMessage =
+    dataSource inTransaction { txn =>
+      val wsNamespace = jobDesc.methodName.workspaceName.namespace
+      val wsName = jobDesc.methodName.workspaceName.name
+      val mcNamespace = jobDesc.methodName.namespace
+      val mcName = jobDesc.methodName.name
+      methodConfigurationDAO.get(wsNamespace, wsName, mcNamespace, mcName, txn) match {
+        case None =>
+          RequestComplete(StatusCodes.NotFound,
+            "No method configuration named "+mcNamespace+"."+mcName+" in workspace "+wsNamespace+"."+wsName)
+        case Some(methodConfig) =>
+          val entType = methodConfig.rootEntityType
+          val entName = jobDesc.entityName
+          entityDAO.get(wsNamespace, wsName, entType, entName, txn) match {
+            case None =>
+              RequestComplete(StatusCodes.NotFound,
+                "No entity of type "+entType+" named "+entName+" in workspace "+wsNamespace+"."+wsName)
+            case Some(entity) =>
+              Try { resolveInputs(methodConfig,entity) } match {
+                case Failure(error) =>
+                  RequestComplete(StatusCodes.Conflict, "Expression evaluation failed: "+error.getMessage())
```

Comment:
it's a conflict between the current state of the resource (an entity that's missing a value, e.g.) and the requirements of the method configuration.
happy to change it to whatever you think makes more sense.